src/base/install/getFilesInDir.m

```diff
@@ -0,0 +1,150 @@
+function files = getFilesInDir(varargin)
+% List all files in the supplied (git tracked) Directory with their absolute path name
+% based on the git ls-file command. If the directory is not git controlled, the
+% gitTypeFlag is assumed to be 'all' and all files (except for .git files
+% will be returned).
+%
+% USAGE:
+%    files = getFilesInDir(varargin)
+%
+% OPTIONAL INPUTS:
+%    varargin:     Options as 'ParameterName',Value pairs. Available
+%                  options are:
+%                  dirToList -  the directory to list the files for, 
+%                               (Default: The current working directory)
+%                  gitTypeFlag - Git type of files to return
+%                                'tracked' - Only tracked files
+%                                'ignored' - Only git ignored files
+%                                            including files that are tracked but
+%                                            ignored (specified in
+%                                            .gitignore). If the folder is
+%                                            not controlled by git, 'all'
+%                                            will be used. 
+%                                'untracked' - anything that is not ignored
+%                                              and not tracked. (new files)
+%                                'all'  - all files except for the git
+%                                         specific files (e.g. .git, .gitignore etc).                               
+%                                'COBRAIgnored' - use the COBRA Toolbox .gitignore
+%                                                 file to determine the ignored files. 
+%                                                 Slower than 'ignored',
+%                                                 since all files have to
+%                                                 be manually checked
+%                                                 against the expressions
+%                                                 specified in the
+%                                                 gitignored file, but
+%                                                 available on all folders
+%                                                 
+%                                (Default: 'all')   
+%                  restrictToPattern - give a regexp pattern to filter the
+%                                      files, this option is ignored if
+%                                      empty. (Default: '', i.e. ignored)
+%                  checkSubFolders - check the subfolders of the current
+%                                    directory. (Default: true)
+% 
+% EXAMPLES:
+%    Get all m files in the source folder:
```

Comment:
May you use here syntax highlighting?

Reply:
how?

Reply:
You would have to use comment style as shown [here](https://github.com/opencobra/cobratoolbox/blob/master/docs/source/guides/documentationGuide.md#keyword-example).